File: java/src/ols2407/math/Complex.java
package ols2407.math;

public class Complex {
    float real;
    float imaginary;

    Complex() {
    }

    Complex(float real, float imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    Complex plus(Complex x) {
        Complex sum = new Complex();
        sum.real = this.real + x.real;
        sum.imaginary = this.imaginary + x.imaginary;
        return sum;
    }

    void diplay() {
        System.out.printf("%1.2f + %1.2fi", real, imaginary);
    }

    public static void main(String[] args) {
        Complex c1 = new Complex(10.1F, 9.8F);
        Complex c2 = new Complex(5.3F, 4.1F);
        Complex c3 = new Complex();
        c3 = c1.plus(c2);
        c3.diplay();
    }
}

File: cpp/class-object/complex.cpp
#include <iostream>

namespace math
{
    class complex
    {
        float real;
        float imaginary;

    public:
        complex()
        {
        }

        complex(float real, float imaginary = 0.0F)
        {
            this->real = real;
            this->imaginary = imaginary;
        }

        complex operator+(complex &x)
        {
            complex sum;
            sum.real = this->real + x.real;
            sum.imaginary = this->imaginary + x.imaginary;
            return sum;
        }

        void diplay()
        {
            std::cout << this->real << " + " << imaginary << "i";
        }
    };
}

int main(int argc, char const *argv[])
{
    math::complex c1(10.1, 9.8), c2(5.3, 4.1), c3;
    c3 = c1 + c2;
    c3.diplay();
    return 0;
}
